autils/archive/archive.py

```
@@ -0,0 +1,539 @@
+        # TAR archives with xz compression
+        ".tar.xz": (False, True, tarfile.open, ":xz"),
+        ".txz": (False, True, tarfile.open, ":xz"),
+        ".xz": (False, True, tarfile.open, ":xz"),
+        # TAR archives with zstd compression
+        ".tar.zst": (False, True, tarfile.open, ":zstd"),
+        ".tzst": (False, True, tarfile.open, ":zstd"),
+        # Standalone compressed files (not tar archives)
+        ".gz": (False, False, None, "gz"),
+        ".bz2": (False, False, None, "bz2"),
+        ".zst": (False, False, None, "zst"),
+    }
```

Ruff (0.16.0)

[warning] Mutable default value for class attribute

(RUF012)